x-pack/plugins/maps/public/actions/store_actions.js

```diff
@@ -458,25 +459,62 @@ export function triggerRefreshTimer() {
   };
 }
 
-export function updateLayerStyleForSelectedLayer(style, temporary = true) {
-  return {
-    type: UPDATE_LAYER_STYLE_FOR_SELECTED_LAYER,
-    style: {
-      ...style,
-      temporary
-    },
+export function clearMissingStyleProperties(layerId) {
+  return async (dispatch, getState) => {
+    const targetLayer = getLayerList(getState()).find(layer => {
+      return layer.getId() === layerId;
+    });
+    if (!targetLayer) {
+      return;
+    }
+
+    const style = targetLayer.getCurrentStyle();
+    if (!style || !style.getDescriptorWithMissingStylePropsRemoved) {
```

Comment:
this type-check, just checking if a method exists, seems kind of clunky.

I would create a new top-class `AbstractStyle` with a dummy implementation, and have heatmap, tile, and vector-style derive from that one

Less involved alternative; you could also check  on `style.type === 'VectorStyle.type'` (you'd need to import VectorStyle module here).

Reply:
I will add an AbstractStyle